packages/api/src/user.js

```diff
@@ -39,9 +40,33 @@ async function loginOrRegister (request, env) {
   const auth = request.headers.get('Authorization') || ''
 
   const token = env.magic.utils.parseAuthorizationHeader(auth)
-  env.magic.token.validate(token)
-
-  const metadata = await env.magic.users.getMetadataByToken(token)
+  const isMagicTestMode = magicTestModeFromEnv(env)
+  let isAllowedTestmodeAuthn = false
+  try {
+    env.magic.token.validate(token)
+  } catch (error) {
+    if (error.code === 'ERROR_INCORRECT_SIGNER_ADDR' && isMagicTestMode) {
```

Comment:
Would it be possible to avoid leaking awareness of magic test mode here? I think we are just supposed to set it as an option on when creating the sdk and then control if it errors or not by what email we use in test code per https://magic.link/docs/auth/introduction/test-mode

Reply:
I reverted just this part to verify that test-e2e can't succeed without this little change. i.e. the tokens that come from using testMode do *not* go through this `magic.token.validate()` fn without throwing.
* test run https://github.com/web3-storage/web3.storage/runs/7887390621?check_suite_focus=true#step:9:19
* playwright-report of fail: https://ipfs.io/ipfs/bafybeif3vlpvh3yqxpwttnlknnc544xluvhvivnumpzs6fqll2lf5qu2iy/trace/index.html?trace=https://ipfs.io/ipfs/bafybeif3vlpvh3yqxpwttnlknnc544xluvhvivnumpzs6fqll2lf5qu2iy/data/dc71c7740419de3f7316bba58bc3d1c6854cf39f.zip

This is in line with my intuition that they mostly shipped testMode for the *frontend* magic sdk instance and in-browser session management stuff.

Reply:
> Would it be possible to avoid leaking awareness of magic test mode here?

Not that I can tell. The test mode stuff really is a special case. *only* when its enabled should we allow an error from the call to `validate`.

I made this commit that encapsulates this logic so this requestHandler is smaller. https://github.com/web3-storage/web3.storage/commit/3505eeb4e44d58159d28ee862279fe626fbcd79f#diff-88d8b13b7d7c1bb4b032ea967283797522a6984ca7424d7ea49bc311b4c82796R84